include/otf2xx/reference.hpp

```diff
@@ -60,29 +62,31 @@ namespace otf2
  *
  * \tparam Type Used to seperate address spaces for different definitions
  */
-template <typename Type>
-class reference
+template <typename Type, typename Tag>
+class reference_impl
 {
 public:
     /**
      * @brief ref_type the underlying type of refernce numbers
      *
      * Mostly uint64_t or uint32_t
      */
-    using ref_type = typename traits::reference_type<Type>::type;
+    using ref_type = typename traits::reference_type<Tag>::type;
```

Comment:
Given this, I'd rename the trait to `reference_from_tag`, remove everything that is not a tag and add `reference_type` as an alias to `reference_from_tag<Definition::tag_type>`

This makes it impossible to instantiate a `reference` with anything that is not explicitly a tag.

Reply:
I just removed them from reference_type. Use `otf2::reference<Def>::ref_type` instead.